### PR TITLE
Deprecate credentials.nickname

### PR DESCRIPTION
**Pull request by nirinchev**
## What, How & Why?

This closes # ???

## ☑️ ToDos

* [x] 📝 Changelog entry
* [ ] 📝 Public documentation PR created or is not necessary

*If this PR adds or changes public API's:*
* [x] typescript definitions file is updated
* [x] jsdoc files updated

### CODE REVIEW

File: lib/index.d.ts [hunk 1]
```diff
@@ -345,7 +345,8 @@ declare namespace Realm.Sync {
         static facebook(token: string): Credentials;
         static google(token: string): Credentials;
         static anonymous(): Credentials;
-        static nickname(value: string, isAdmin?: boolean): Credentials;
+        /** @deprecated will be removed in a future ROS version. Use usernamePassword or anonymous instead. */
```
Comment (kneth) [inline lib/index.d.ts]:
```suggestion
        /** @deprecated will be removed in a future version. Use usernamePassword or anonymous instead. */
```

File: lib/user-methods.js [hunk 2]
```diff
@@ -421,9 +421,9 @@ const credentialsMethods = {
         return new Credentials('anonymous');
     },
 
-    nickname(value, isAdmin) {
-        checkTypes(arguments, ['string', 'boolean']);
-        return new Credentials('nickname', value, { is_admin: isAdmin || false });
+    nickname(value) {
+        checkTypes(arguments, ['string' ]);
```
Comment (kneth) [inline lib/user-methods.js]:
```suggestion
        checkTypes(arguments, ['string']);
```

File: tests/js/download-api-helper.js [hunk 3]
```diff
@@ -32,19 +32,11 @@ function createObjects(user) {
...
-    return new Promise((resolve, reject) => {
-        let callback = (transferred, total) => {
-            if (transferred === total) {
-                session.removeProgressNotification(callback);
-                resolve(realm);
-            }
-        }
-        session.addProgressNotification('upload', 'forCurrentlyOutstandingWork', callback);
-    });
+    return realm.syncSession.uploadAllLocalChanges();
```
Comment (kneth) [inline tests/js/download-api-helper.js]: Nice simplification!

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,9 @@
+3.0.0-beta.X Release notes (TBD)
+=============================================================
+### Breaking Changes
+* Removed the `isAdmin` parameter from `Credentials.nickname` as that has no effect in new ROS versions.
```
Comment (kneth) [inline CHANGELOG.md]:
```suggestion
* Removed the `isAdmin` parameter from `Credentials.nickname` as that has no effect in new Realm Object Server versions.
```

Comment (kneth) [reply CHANGELOG.md]: Do you remember which ROS version?

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,9 @@
...
+* Deprecated the `Credentials.nickname` method as the nickname auth provider will not be supported in future ROS versions.
```
Comment (kneth) [inline CHANGELOG.md]:
```suggestion
* Deprecated the `Credentials.nickname` method as the nickname auth provider will not be supported in future Realm Object Server versions.
```

File: docs/sync.js [hunk 6]
```diff
@@ -457,10 +457,10 @@ class Credentials {
      * Creates credentials based on a login with a nickname. If multiple users try to login
      * with the same nickname, they'll get the same underlying sync user.
      * @param {string} value The nickname of the user.
-     * @param {boolean} [isAdmin] An optional parameter controlling whether the user is admin. Default is `false`.
      * @return {Credentials} An instance of `Credentials` that can be used in {@linkcode Realm.Sync.User.login|User.login}.
+     * @deprecated will be removed in a future ROS version. Use usernamePassword or anonymous instead.
```
Comment (kneth) [inline docs/sync.js]:
```suggestion
     * @deprecated will be removed in a future ROS version. Use {@linkcode Realm.Sync.Credentials.usernamePassword} or {@linkcode Realm.Sync.Credentials.anonymous} instead.
```

File: CHANGELOG.md (hunk 4 above)
Comment (nirinchev) [reply CHANGELOG.md]: In the next ROS version 😄

**Pull request merged**